### PR TITLE
feat(InlineContent): Font Strikethrough

### PR DESCRIPTION
**Pull request by shanteaustin**
## Description

Updated text styles to include an option to strike through 

## Testing

Go to Inline Content --> With Strike Through 
To view line through text

## Checklist

- [X] all commented code has been removed
- [X] any new console issues have been resolved
- [X] code linter and formatter has been run
- [X] test coverage meets repo requirements
- [X] PR name matches the expected semantic-commit syntax

### CODE REVIEW

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: This can be removed since this style isn't being used in this story.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: I think we're going to have to workshop this. It seems to be causing the text after it to overlap when `contentWrap` is true.

Screenshot 2024-09-12 at 4 02 49 PM

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: Removed

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (cee-money) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: the `h` might need to be dynamic (a percentage of the text component's h). for example what if the font size is 40? the strikethrough might appear too thin.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (cee-money) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: we may want to add `mountY: 0.5` so that the strikethrough is fully vertically centered in the text

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Great point

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: The mountY doesnt work on my end to center. Did you test this out?

Comment (cee-money) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: i did, it's definitely minor 😆

but here's what i'm seeing. `mountY: 1` might be best? 
Screenshot 2024-09-13 at 11 23 58 AM

Comment (cee-money) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: or potentially something like
```
h: textComponent.h * 0.08,
y: textComponent.h / 2 - textComponent.h * 0.08,
mountY: 0.5
```

Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Just to throw my nitpicking two cents in, I think this looks best:
```js
h: textComponent.h * 0.08,
y: textComponent.h / 2,
mountY: 1
```

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Adding `h: textComponent.h` here will resolve the overlapping issue when contentWrap is true.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:1]: We're going to have to update the announce method to accommodate for the strikethrough. I posed the question to the team about whether "strikethrough" should come before or after the text it is applied to, but essentially, this will need to be update in `announce()`:

```js
else if (item.text) {
  announce += item.text;

  if (item.style?.textDecoration === 'line-through') {
    announce += 'strikethrough';
  }
}
```

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Looks perfectly centered now. Thanks guys!

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js
Comment (cee-money) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: i think a trailing space sneaked in here

```suggestion
              text: '$19.99',
```

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (cee-money) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:542]: just checking, is override-able? @ImCoolNowRight i vaguely remember this coming up during our last sync. did we decide that users should be able to pass whatever announce text they want into this like: "Price used to be __ but is now __"?

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: 1. the width and the height are both being calculated with some hardcoded values, are we cool with that?
2. should the fallback color be derived from a theme value?

Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: 1. @shanteaustin Do we need the 2 being subtracted? I think the height is a good callout, we can pull the 0.08 out into a style.
2. I don't think it hurts to point to a token by default. At least the strikethrough would be in line with the user's theme.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:1]: The if statement on 104 should be updated to:

```js
if (
  (nextItem && isText(nextItem)) ||
  (this.contentWrap &&
    nextItem &&
    nextItem.newline &&
    this._parsedContent[index + 2] &&
    isText(this._parsedContent[index + 2]))
  ) {
```

As it stands right now, the spacing is off on some lines if you set `contentWrap` to true. This, along with the other comment in this review, will make sure that everything is spaced out properly.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: This needs to be updated to `w: textWidth + textComponent.flexItem.marginRight,`

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:542]: Still waiting on confirmation from the accessibility team. Personally, I think this is a good generic default, and that clients can override `_announce` for their needs on a case by case basis. But we'll see what they say!

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: If the space is not added the text is too close together. If you take a look at the file some of the text have spaces added.

Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: But I will remove it and push up. It may look good to you without the added space.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: 1. @ImCoolNowRight The line seemed at bit long to me but after a couple tweaks it's not so bad so I will remove that. When you say pull the 0.08 into a style do you mean adding it to the theme object in InlineContent.styles? 
2. Which token should we use? textNeutral?

Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: 1. We have a `.styles.js` file for each component that has the styles used by it. We're suggesting adding a new style property to it, something like `strikethroughRatio`, that would be set to 0.08 by default, and then we reference that style instead of the hard coded number. That way, users can pass something else in if they want to.
2. I think textNeutral works!

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:1]: Will update as suggested.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js
Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.stories.js]: I think this space is the only thing left to change.

**Review by ImCoolNowRight — COMMENTED** [review]
Just need to remove that extra space in the story, and then I think this will be good to go! I'm ok with circling back once we hear back about how to handle the announcer.

**Comment by shanteaustin** [comment]
> Just need to remove that extra space in the story, and then I think this will be good to go! I'm ok with circling back once we hear back about how to handle the announcer.

I thought that was remove looks like it was reverted so I updated it again.

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Sorry, one last thing!

Can we actually reverse this order and rename the style property to `strikethroughColor` (and update the style file, too)? I think that way makes more sense so that users can define a color they want the strikethrough to be, and then we provide fallbacks.

So I think this should be `color: this.style.strikethroughColor || textOverrideStyles?.textColor || this.style.textStyle.textColor`. That'll fall back to the text override first, and then the default text style.

Comment (shanteaustin) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js]: Sounds good. I just updated the property name and changed the order.

**Review by ImCoolNowRight — APPROVED** [review]

**Review by cee-money — APPROVED** [review]
LGTM

File: packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js
Comment (ImCoolNowRight) [reply packages/@lightningjs/ui-components/src/components/InlineContent/InlineContent.js:542]: Resolving this conversation to get this merged in. Will create a follow-up ticket to ensure this gets completed.

**Pull request merged**
